spring-cloud-stream-app-starters-docs/src/main/asciidoc/overview.adoc

```diff
@@ -492,4 +494,146 @@ Then do this:
+ We need this bom during app generation to pull down all the core dependencies.
+
+. What are the contents of the core BOM? +
+In addition to the common artifacts in core, the app-starters-core-dependencies BOM also adds dependency management for spring-cloud-dependencies which will include spring-cloud-stream transitively.
+
+. Where is the core BOM used? +
+There are two places where the core BOM is used. It is used to provide compile time dependency management for all the starters.
+This is defined in the `app-starters-build` artfiact. This same BOM is referenced through the maven plugin configuration for the app generation.
+The generated apps thus will include this bom also in their pom.xml files.
+
+. What are the artifacts that the parent artifact (`app-starters-build`) include from the core bom to be part of all the starters by default and there by making them part of the generated apps? +
+* spring-cloud-stream
+* Spring-cloud-stream-test-support-internal
+* spring-cloud-stream-test-support
+
+. What other artfiacts are available through the parent `app-starters-build` and where are they coming from? +
+In addition to the above artifacts, the artifacts below also included in `app-starters-build` by default.
+* json-path
+* spring-integration-xml
+* spring-boot-starter-logging
+* spring boot-starter-security +
+Spring-cloud-build is the parent for app-starters-build. Spring-cloud-build imports spring-boot-dependencies and that is from where these artifacts are coming from.
+
+. I did not see any other Spring Integration components used in the above 2 lists. Where are those dependencies coming from for individual starters? +
+Spring-integration bom is imported in the spring-boot-dependencies bom and this is where the default SI dependencies are coming for SCSt app starters.
+
+. Can you summarize all the BOM's that SCSt app starters depend on? +
+All SCSt app starters have access to dependencies defined in the following BOM’s and other dependencies from any other BOM’s these three boms import transitively as in the case of Spring Integration:
+    * app-starters-core-dependencies
+    * spring-cloud-dependencies
+    * spring-boot-dependencies
+
+.  Each app starter has `app-starter-build` as the parent which in turn has `spring-cloud-build` as parent. The above documentation states that the
+generated apps have `spring-boot-starter` as the parent. Where is the mismatch? +
+There is no mismatch per se, but a slight subtlety. As the question frames, each app starter has access to artifacts managed all the way through `spring-cloud-build` at compile time.
+However, this is not the case for the generated apps at runtime. Their parent is `spring-boot-starter` that imports `spring-boot-dependencies` bom that includes a majority of the components that these apps need.
+The question then is what if the starter used an artifact managed through the `app-starter-build` or `spring-cloud-build`. Remember that the
```

Comment:
all the dependencies coming from `spring-boot-dependencies` are good here. I was talking about the dependencies specified only at the spring-cloud-dependencies` bom or our specific `core-app-starter` bom. I will try to clarify this in the answer.